### PR TITLE
Added support for xlsx files

### PR DESCRIPTION
**Pull request by kleppari**
- Support for reading xlsx files using openpyxl

### CODE REVIEW

File: clarity_ext/context.py [hunk 1]
```diff
@@ -221,7 +221,8 @@ def handle_validation(self, msg=None):
         self.validation_service.handle_validation(
             self.validation_results, tailored_error_message=msg)
 
-    def local_shared_file(self, clarity_file_handle, mode="r", is_xml=False, is_csv=False, file_name_contains=None):
+    def local_shared_file(self, clarity_file_handle, mode="r", is_xml=False, is_csv=False, file_name_contains=None,
```
Comment (Aratz) [inline clarity_ext/context.py:224]: I know the body of this function is old code, but wouldn't it be better to remove the extra `is_*` arguments and only rely on detecting the file type automatically? It seems the bulk of what this function is doing is handling the case where the flag and the file type don't agree.

So I'd suggest either remove all the flags and detect the file type by looking at the extension in the file name, or, if it's important to have the user specify the file type they expect, to replace the flag args by `extension` where one could pass a string e.g. 'csv'

File: clarity_ext/context.py [hunk 2]
```diff
@@ -246,6 +247,9 @@ def check_file_extension(extension):
         elif is_csv:
             check_file_extension('.csv')
             return self.file_service.parse_csv(f)
+        elif is_xlsx:
+            check_file_extension('.xlsx')
```
Comment (Aratz) [inline clarity_ext/context.py:251]: Can it happen that the extension of the sample form is just `.xls`?

File: clarity_ext/service/file_service.py [hunk 3]
```diff
@@ -58,6 +58,16 @@ def __init__(self, artifact_service, file_repo, should_cache, os_service, upload
...
+    def parse_xlsx(self, f):
+        from openpyxl import load_workbook
+        import pandas as pd
+        with open(f.name, "rb") as wbook:
+            writer = pd.ExcelWriter(wbook, engine='openpyxl')
+            workbook = load_workbook(wbook, data_only=True)
+
+            writer.book = workbook
+            return writer.book
+
```
Comment (Aratz) [inline clarity_ext/service/file_service.py]: I tried this locally but I get an `AttributeError: can't set attribute` when reaching `writer.book = workbook` :confused: 

Is `pd.ExcelWriter` necessary here or can you return `workbook` immediately?

**Review by Aratz — CHANGES_REQUESTED** [review]
Looks good!

My main question is regarding the `local_shared_file` function, it feels like maybe it could benefit from a rewrite.

The other main question is with the `parse_xlsx` function, I tried it locally but couldn't really make it work.

File: clarity_ext/context.py (hunk 1 above)
Comment (kleppari) [reply clarity_ext/context.py:224]: I will create a ticket for this

File: clarity_ext/context.py (hunk 2 above)
Comment (kleppari) [reply clarity_ext/context.py:251]: No, the sample from is just .xlsx

**Review by Aratz — APPROVED** [review]
Looks good, although I think you need to add `openpyxl` to the requirements for the tests to pass.

**Pull request merged**